### PR TITLE
avatar root slot

### PR DESCRIPTION
**Pull request by DoubleStyx**

### CODE REVIEW

File: Logix/Slots/GetNthParent.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using FrooxEngine.LogiX.Math;
+using BaseX;
+using FrooxEngine.CommonAvatar;
+
+
+namespace FrooxEngine.LogiX.Slots
+{
+    [Category("LogiX/Slots")]
+    [NodeName("Get Nth Parent")]
+    public class GetNthParent: LogixOperator<Slot>
```
Comment (Frozenreflex) [inline Logix/Slots/GetNthParent.cs:13]: This is a duplicate of GetGrandparent.

**Review by Frozenreflex — CHANGES_REQUESTED** [review]

File: Logix/Slots/GetNthParent.cs (hunk 1 above)
Comment (DoubleStyx) [reply Logix/Slots/GetNthParent.cs]: Acknowledged; removed this from the pr.

**Review by Frozenreflex — APPROVED** [review]

File: Logix/Avatar/AvatarRootSlot.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public readonly Input<User> User;
+        public override Slot Content
+        {
+            get
+            {
+                User user = User.Evaluate();
+                if (user == null)
+                    return null;
+                Slot slot = user.Root.Slot;
+                foreach (var item in slot.GetAllChildren())
```
Comment (ProbablePrime) [inline Logix/Avatar/AvatarRootSlot.cs:24]: Slot.GetComponentsInChildren<AvatarRoot> ?

Comment (ProbablePrime) [reply Logix/Avatar/AvatarRootSlot.cs:24]: GetFirstDirectComponentsInChildren probably actually, that stops the search when it finds one.

Comment (Xlinka) [reply Logix/Avatar/AvatarRootSlot.cs]: Resolving this as they seemed to have done what you said, nice to see you helping ^_^

**Pull request merged**